cmake/add_performance_test.cmake

```diff
@@ -0,0 +1,116 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Add a google benchmark test which utilizes performance_test_fixture to
+# leverage osrf_testing_tools_cpp.
+#
+# Call add_executable(target ARGN), link it against the google benchmark
+# libraries and register the executable as a test.
+#
+# If google benchmark is not available the specified target is not being created
+# and therefore the target existence should be checked before being used.
+#
+# If osrf_testing_tools_cpp memory_tools is not available the test is skipped.
+#
+# :param target: the target name which will also be used as the test name
+# :type target: string
+# :param ARGN: the list of source files
+# :type ARGN: list of strings
+# :param RUNNER: the path to the test runner script, default defined by
+#   ``ament_add_test()``).
```

Comment:
@cottsay nit: 
```suggestion
#   ``ament_add_test()``
```

I know you took it from another place. We should still try to improve the codebase if we can.